src/components/Input/Input.tsx

```diff
@@ -34,16 +38,27 @@ const Input: React.FC<InputProps> = ({
     value = '',
     width = '320px',
     labelBgColor,
+    isWalletAddress,
 }: InputProps) => {
+    const inputEl = useRef<HTMLInputElement>(
+        null,
+    ) as MutableRefObject<HTMLInputElement>;
+
     const [currentValue, setCurrentValue] = useState(value);
     const [currentState, setCurrentState] = useState(state);
     const [mainType, setMainType] = useState(type);
     const [isInputHidden, setIsInputHidden] = useState(inputHidden);
     const [invalidMessage, setInvalidMessage] = useState(errorMessage);
+    const [isEditMode, setIsEditMode] = useState(true);
 
     useEffect(() => setIsInputHidden(type === 'password'), [inputHidden]);
     useEffect(() => setCurrentState(state), [state]);
     useEffect(() => setMainType(type), [type]);
+    useEffect(() => {
```

Comment:
this is cool and it helps with your input, but checkout the other stories, especially checkout Form... maybe it is worth considering to make a new component like you said, this one already has a lot of logic.

If the new component called in Input with type text and Blockie, would it not be easier to control their state from a parent without effecting their core functionality perhaps?

Reply:
I think it is not doable with the current input structure as there is no shared functionality, because here we need to edit the input to look and behave in a certain way so it's not possible to add on top of it.

Reply:
can you explain that more for me sorry 😅 

Input onChange will return the value, can you not store that string in a parent component and feed it to the Blockie compoent?

Reply:
Thr blockie is inside the input and it should replace the prefix so how could I add blockie if I grap the normal Input
That will require changes to the Input field to accept components for the prefix so it could then accept blockie and put it there

That previous isn't the problem, the problem occurs when you want to display ellipsis text I have then to hide the input field(not all the component but the input itself) and display the ellipse text on blur and on focus I have to revert back to show the input field so then this could not be done by any mean if I just import the Input Component 

I hope this will help you understand the approach

Reply:
I would I hack it with css while I can create separate components and reuse the same styles. We just need to remove the code duplicates, but that will be later on when we restructure the input better.

"Sorry i dont know what you mean here, do you have an example of any screen shots? TY"
if you take a look on the design of the input you will find the text is ellipsised 
<img width="363" alt="image" src="https://user-images.githubusercontent.com/13832392/162477857-ed84f0f1-8d77-45f4-ba32-91d26c3a971b.png">
in the first glance you will think easy just make the text ellipsis and put it in the input value. But think of a use case where someone is using it in a form and you submit the form, what value you will get? the ellipsised wallet address not the actual wallet address
So the case here is you only need to display wallet address ellipsis without changing the input value, and the solution to that is to create a text element that holds the ellipsis text and hide the input and when the user want to edit hide that element and show the input back
that would then end up like this 
![wallet input](https://user-images.githubusercontent.com/13832392/162018654-b7d4452e-8747-4294-9e7d-0b7527a3b780.gif)

I believe you haven’t checked the code yet so I encourage you to see it and see the recent commits on this PR where I separate the components while having the same styles.

Reply:
ah nice i had such a busy day, this looks cool, i'll take some item tomorrow to review it again for you